test/test_suite_runner.js

```diff
@@ -324,6 +324,29 @@ describe('Suite runner', function() {
     return shouldFail(runTestSuite('suite_single_only_test', [], { disallowOnly: true }));
   });
 
+  it('should detect and kill orphan childprocess', function() {
+    var proc = require('child_process').fork('test/util/run_single_test_that_never_finishes.js');
+    return new Promise(function(resolve, reject) {
+      setTimeout(function() {
+        require('ps-tree')(process.pid, function(err, children) {
+          var childrenPID = children.map(function(p) { return p.PID; });
+          // Check that the childprocesses has killed itself within 2 sec
+          setTimeout(function() {
+            var livingChildProcesses = [];
+            childrenPID.forEach(function(childPID) {
+
+              if (require('is-running')(childPID)) {
+                livingChildProcesses.push(childPID);
+              }
+            });
+            (livingChildProcesses.length > 0)? reject('child process is still alive: ' + livingChildProcesses): resolve();
+          }, 2000);
```

Comment:
You may not need to have a timeout here: Another option is to simply make the test run until the child process dies. Then it will fail with a timeout from the test runner if it doesn't work.